lib/semantics/check-omp-structure.h

```diff
@@ -12,24 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// OpenMP clause validity check for directives
+// OpenMP structure validity check list
+//    1. invalid clauses on directive
+//    2. invalid repeated clauses on directive
+//    3. TODO: invalid nesting of regions
 
 #ifndef FORTRAN_SEMANTICS_CHECK_OMP_STRUCTURE_H_
 #define FORTRAN_SEMANTICS_CHECK_OMP_STRUCTURE_H_
 
 #include "semantics.h"
+#include "../common/enum-set.h"
 #include "../parser/parse-tree.h"
-#include <typeindex>
 
 namespace Fortran::semantics {
 
+ENUM_CLASS(OmpDirectiveEnum, PARALLEL, DO, SECTIONS, SECTION, SINGLE, WORKSHARE,
```

Comment:
I think `OmpDirective` is a better name.

Reply:
Addressed in commit [cb89744](cb89744746186bcf120d8ca62a791e2778366b9b).